packages/jellyfish-api-core/__tests__/category/account/listPendingFutureSwaps.test.ts

```diff
@@ -0,0 +1,570 @@
+        await testing.generate(1)
+
+        await testing.container.call('futureswap', [tslaAddress, '0.2@DUSD', 'TSLA'])
+        await testing.generate(1)
+
+        // Call listpendingfutureswaps after performing futureswap
+        const pendingFutureSwaps1 = await testing.rpc.account.listPendingFutureSwaps()
+        expect(pendingFutureSwaps1).toStrictEqual(
+          [
+            {
+              owner: tslaAddress,
+              source: '0.20000000@DUSD',
+              destination: 'TSLA'
+            },
+            {
+              owner: tslaAddress,
+              source: '0.40000000@DUSD',
+              destination: 'TSLA'
+            },
+            {
+              owner: tslaAddress,
+              source: '0.60000000@DUSD',
+              destination: 'TSLA'
+            },
+            {
+              owner: tslaAddress,
+              source: '0.80000000@DUSD',
+              destination: 'TSLA'
+            }
+          ]
+        )
+
+        // Wait for 1 block before getfutureswapblock
+        const nextSettleBlock = await testing.container.call('getfutureswapblock')
+        await testing.generate(nextSettleBlock - await testing.rpc.blockchain.getBlockCount() - 1)
+
+        const pendingFutureSwaps2 = await testing.rpc.account.listPendingFutureSwaps()
+        expect(pendingFutureSwaps2).toStrictEqual(pendingFutureSwaps1) // Nothing change
+
+        // Generate 1 more block to futureswap
+        await testing.generate(1)
+
+        // Call listpendingfutureswaps after futureswap
+        {
+          const pendingFutureSwaps = await testing.rpc.account.listPendingFutureSwaps()
+          expect(pendingFutureSwaps).toStrictEqual([])
+        }
+      })
+    })
+
+    describe('For multiple tokens which are TSLA, AAPL, MSFT', () => {
+      it('Should listPendingFutureSwaps if multiple futureswaps TSLA, AAPL, MSFT for DUSD', async () => {
+        // Call listpendingfutureswaps before performing futureswap
+        {
+          const pendingFutureSwaps = await testing.rpc.account.listPendingFutureSwaps()
+          expect(pendingFutureSwaps).toStrictEqual([])
+        }
+
+        // futureswap 2 times for every token
+        await testing.container.call('futureswap', [tslaAddress, '0.85@TSLA'])
+        await testing.container.call('futureswap', [tslaAddress, '0.15@TSLA'])
+        await testing.container.call('futureswap', [aaplAddress, '0.71@AAPL'])
+        await testing.container.call('futureswap', [aaplAddress, '0.29@AAPL'])
+        await testing.container.call('futureswap', [msftAddress, '0.63@MSFT'])
+        await testing.container.call('futureswap', [msftAddress, '0.27@MSFT'])
+        await testing.generate(1)
+
+        // Call listpendingfutureswaps after performing futureswap
+        const pendingFutureSwaps1 = await testing.rpc.account.listPendingFutureSwaps()
+        expect(pendingFutureSwaps1.length).toStrictEqual(6)
+
+        expect(pendingFutureSwaps1).toStrictEqual(
```

Comment:
As of this writing, listPendingFutureSwaps returns elements ordered by txid. As txid is a random value, I am unable to use toStrictEqual here.